Source/SwiftLintFramework/Rules/Idiomatic/ExplicitInitRule.swift

```diff
@@ -164,52 +163,78 @@ public struct ExplicitInitRule: SubstitutionCorrectableASTRule, ConfigurationPro
         ]
     )
 
-    public func validate(file: SwiftLintFile, kind: SwiftExpressionKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        return violationRanges(in: file, kind: kind, dictionary: dictionary).map {
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: $0.location))
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor()
+    }
+
+    public func makeRewriter(file: SwiftLintFile) -> ViolationsSyntaxRewriter? {
+        file.locationConverter.map { locationConverter in
+            Rewriter(
+                locationConverter: locationConverter,
+                disabledRegions: disabledRegions(file: file)
+            )
         }
     }
+}
 
-    private let initializerWithType = regex(#"^([A-Z][^(\s]*)\s*\.init$"#)
+private extension ExplicitInitRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            guard
+                let calledExpression = node.calledExpression.as(MemberAccessExprSyntax.self),
+                let calledBase = calledExpression.base?.as(IdentifierExprSyntax.self),
+                calledBase.identifier.text.first?.isUppercase == true,
+                calledExpression.name.text == "init",
+                let violationPosition = calledExpression.base?.endPositionBeforeTrailingTrivia
+            else {
+                return
+            }
 
-    public func violationRanges(in file: SwiftLintFile, kind: SwiftExpressionKind,
-                                dictionary: SourceKittenDictionary) -> [NSRange] {
-        guard kind == .call,
-              let name = dictionary.name,
-              let typeRange = findTypeRange(in: name),
-              let nameByteRange = dictionary.nameByteRange else {
-            return []
+            violationPositions.append(violationPosition)
```

Comment:
```suggestion
            violationPositions.append(calledBase.endPositionBeforeTrailingTrivia)
```
Should be possible, right?

Reply:
Yes, will do.